### ISSUE TITLE
FR: Add option to rotate book

### ISSUE THREAD

**Issue by bahusoid** [koreader/koreader#3280]
I have some pdf and djvu books scanned vertically (and some even upside down). Now I can read them in portrait mode but with reader placed in landscape. That makes page turns and menu navigation awkward. Reflow doesn't work for them also.

It would be cool to have option to rotate pages (at least for pdf and djvu). This rotation should be applied before Reflow, OCR or any other post processing options. It would allow to read comfortably books that are improperly scanned. This rotation should be persistent for all pages and be applied on next open.

For instance in bottom menu along with Orientation option can also be added Rotate option with two buttons Right and Left. Each press will rotate image 90 degree  to the right/left. 

[VerticallyScannedBookExample.pdf](https://github.com/koreader/koreader/files/1346631/VerticallyScannedBookExample.pdf)

**Comment by Frenzie** [comment]
While for the best results I think you should use ScanTailor (you can read about a little wrapper script I wrote [here](http://fransdejonge.com/2014/10/fixing-up-scanned-pdfs-with-scan-tailor/)), I'd strongly recommend tossing scanned PDFs in a program like PDF-Shuffler, PDF Scissors or pdfsandwich.

PDF-Shuffler is currently my favorite tool for quick GUI PDF manipulations.

![screenshot_2017-09-30_20-47-12](https://user-images.githubusercontent.com/202757/31048595-9119fffa-a620-11e7-91c3-590a31fccc19.png)

Naturally I welcome someone adding it, but to me doing such document manipulations on the reader is somewhat cumbersome at the best of times, even while gracefully forgetting about those moments when its CPU is woefully inadequate.

**Comment by bahusoid** [comment]
Yeah I know how to fix them. I personally use http://www.willus.com/k2pdfopt/

Which supports both pdf and djvu. And allows quickly to fix books via console. For instance with my example book on Windows:
```
:: -mode copy = to skip any processing,  -n = native pdf, -rt 90 = rotate 90 degree.
k2pdfopt.exe -mode copy -n -rt 90 VerticallyScannedBookExample.pdf
```

It's just current reflow function works pretty cool if book is correctly rotated. So no tools would be needed if this feature is implemented.

**Comment by Frenzie** [comment]
Oh, why didn't you say so before? That makes it something anybody *cough*you? ;-)*cough* could probably whip up in fairly little time. All that'd be needed is to write a little bit of GUI frontend code to expose the k2pdfopt backend.

https://github.com/koreader/koreader-base/blob/7a5ca100317ad61fdb6ce84f95e20991dc78816e/ffi/koptcontext.lua#L31
